src/main/java/org/opensearch/knn/index/query/KNNQueryFactory.java

```diff
@@ -86,10 +89,12 @@ public static Query create(CreateQueryRequest createQueryRequest) {
             return new KNNQuery(fieldName, vector, k, indexName);
         }
 
+        log.debug(String.format("Creating Lucene k-NN query for index: %s \"\", field: %s \"\", k: %d", indexName, fieldName, k));
+        BitSetProducer parentFilter = createQueryRequest.context == null ? null : createQueryRequest.context.getParentFilter();
```

Comment:
who is setting this parent filter in the CreateQueryRequest?

Reply:
https://github.com/opensearch-project/OpenSearch/pull/10246